sycl/include/sycl/ext/oneapi/experimental/cuda/non_uniform_algorithms.hpp

```diff
@@ -24,67 +24,67 @@ using IsRedux = std::bool_constant<
 //// Masked reductions using redux.sync, requires integer types
 
 template <typename Group, typename T, class BinaryOperation>
-std::enable_if_t<
-    is_sugeninteger<T>::value && IsMinimum<T, BinaryOperation>::value, T>
+std::enable_if_t<is_sugeninteger_v<T> && IsMinimum<T, BinaryOperation>::value,
+                 T>
 masked_reduction_cuda_sm80(Group g, T x, BinaryOperation binary_op,
                            const uint32_t MemberMask) {
   return __nvvm_redux_sync_umin(x, MemberMask);
 }
 
 template <typename Group, typename T, class BinaryOperation>
-std::enable_if_t<
-    is_sigeninteger<T>::value && IsMinimum<T, BinaryOperation>::value, T>
+std::enable_if_t<is_sigeninteger_v<T> && IsMinimum<T, BinaryOperation>::value,
+                 T>
 masked_reduction_cuda_sm80(Group g, T x, BinaryOperation binary_op,
                            const uint32_t MemberMask) {
   return __nvvm_redux_sync_min(x, MemberMask);
 }
 
 template <typename Group, typename T, class BinaryOperation>
-std::enable_if_t<
-    is_sugeninteger<T>::value && IsMaximum<T, BinaryOperation>::value, T>
+std::enable_if_t<is_sugeninteger_v<T> && IsMaximum<T, BinaryOperation>::value,
+                 T>
 masked_reduction_cuda_sm80(Group g, T x, BinaryOperation binary_op,
                            const uint32_t MemberMask) {
   return __nvvm_redux_sync_umax(x, MemberMask);
 }
 
 template <typename Group, typename T, class BinaryOperation>
-std::enable_if_t<
-    is_sigeninteger<T>::value && IsMaximum<T, BinaryOperation>::value, T>
+std::enable_if_t<is_sigeninteger_v<T> && IsMaximum<T, BinaryOperation>::value,
+                 T>
 masked_reduction_cuda_sm80(Group g, T x, BinaryOperation binary_op,
                            const uint32_t MemberMask) {
   return __nvvm_redux_sync_max(x, MemberMask);
 }
 
 template <typename Group, typename T, class BinaryOperation>
-std::enable_if_t<(is_sugeninteger<T>::value || is_sigeninteger<T>::value) &&
-                     IsPlus<T, BinaryOperation>::value,
+std::enable_if_t<(is_sugeninteger_v<T> ||
+                  is_sigeninteger_v<T>)&&IsPlus<T, BinaryOperation>::value,
                  T>
 masked_reduction_cuda_sm80(Group g, T x, BinaryOperation binary_op,
                            const uint32_t MemberMask) {
   return __nvvm_redux_sync_add(x, MemberMask);
 }
 
 template <typename Group, typename T, class BinaryOperation>
-std::enable_if_t<(is_sugeninteger<T>::value || is_sigeninteger<T>::value) &&
-                     IsBitAND<T, BinaryOperation>::value,
+std::enable_if_t<(is_sugeninteger_v<T> ||
```

Comment:
I know this wasn't your doing, but could we add the parentheses so that   xxx && yyy || zz   only has one clear  interpretation?  I believe it's  (xx && yy) || zz, but to be certain I have to go look up the C++ precedence rules for these operators.

Reply:
I'd rather merge it as-is since the testing has finished and I'd want to be able to upload further PRs based on this one.